packages/material/src/complex/MaterialTableControl.tsx

```diff
@@ -202,60 +209,115 @@ interface NonEmptyRowProps {
-  ));
+const NonEmptyRow = React.memo(
+  ({
+    childPath,
+    schema,
+    rowIndex,
+    openDeleteDialog,
+    moveUp,
+    moveDown,
+    isLast,
+    sortButtons
+  }: NonEmptyRowProps & WithDeleteDialogSupport) => {
+    return (
+      <TableRow key={childPath} hover>
+        {generateCells(NonEmptyCell, schema, childPath)}
+        <NoBorderTableCell
+          style={sortButtons ? styles.fixedCell : styles.fixedCellSmall}
+        >
+          <div style={{ display: 'flex', justifyContent: 'center' }}>
+            <Grid
+              container
+              direction="row"
+              justify="center"
+              alignItems="center"
+            >
+              {sortButtons ? (
+                <Fragment>
+                  <Grid item>
+                    <IconButton
+                      aria-label={`move up`}
+                      onClick={moveUp}
+                      disabled={rowIndex == 0}
+                    >
+                      <ArrowUpward />
+                    </IconButton>
+                  </Grid>
+                  <Grid item>
+                    <IconButton
+                      aria-label={`move down`}
+                      onClick={moveDown}
+                      disabled={isLast}
+                    >
+                      <ArrowDownward />
+                    </IconButton>
+                  </Grid>
+                </Fragment>
+              ) : (
+                ''
+              )}
+
+              <Grid item>
+                <IconButton
+                  aria-label={`Delete`}
+                  onClick={() => openDeleteDialog(childPath, rowIndex)}
+                >
+                  <DeleteIcon />
+                </IconButton>
+              </Grid>
+            </Grid>
+          </div>
+        </NoBorderTableCell>
+      </TableRow>
+    );
+  }
+);
 interface TableRowsProp {
   data: number;
   path: string;
   schema: JsonSchema;
+  moveUp?(path: string, toMove: number): () => any;
+  moveDown?(path: string, toMove: number): () => any;
+  uischema: ControlElement;
 }
 const TableRows = ({
   data,
   path,
   schema,
-  openDeleteDialog
+  openDeleteDialog,
+  moveUp,
+  moveDown,
+  uischema
 }: TableRowsProp & WithDeleteDialogSupport) => {
   const isEmptyTable = data === 0;
-
   if (isEmptyTable) {
     return <EmptyTable numColumns={getValidColumnProps(schema).length + 1} />;
   }
-
   return (
     <React.Fragment>
       {range(data).map((index: number) => {
         const childPath = Paths.compose(
           path,
           `${index}`
         );
-
         return (
           <NonEmptyRow
             key={childPath}
             childPath={childPath}
             rowIndex={index}
             schema={schema}
             openDeleteDialog={openDeleteDialog}
+            moveUp={moveUp(path, index)}
+            moveDown={moveDown(path, index)}
+            isLast={index === data - 1}
+            sortButtons={uischema.options && uischema.options.sortButtons}
```

Comment:
for the showing of the buttons we should also take the enablement state of the whole table into account, as we probably don't want to show the buttons if the table is not editable

Reply:
@eneufeld  here you mean to have a check based on the `edited` prop? 
Why I am asking this is, because I could not see any similar check for the display of delete button. Moreover, there is no logic in this component based on that particular prop.

Reply:
Yes, this is missing which is a bug as we cannot disable the table atm

Reply:
can we skip the particular suggestion for now? This can be solved when we solve the bug.